src/scopeguard.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <functional>
```

Comment:
yay! I've been craving a scope guard but was too lazy to do it myself. Is this implementation mostly copied from std::experimental::scope_exit ?

Reply:
No, I wrote this